packages/mobile/src/fiatExchanges/saga.ts

```diff
@@ -121,52 +117,59 @@ function* bidaliPaymentRequest({
   }
 }
 
-export function* searchNewItemsForProviderTxs({ transactions }: NewTransactionsInFeedAction) {
+export function* fetchTxHashesToProviderMapping() {
+  const account = yield call(getAccount)
+  const channel = yield call(providerTxHashesChannel, account)
+
+  try {
+    if (!channel) {
+      throw new Error('Unable to read from providerTxHashesChannel')
+    }
+
+    const txHashesToProvider: TxHashToProvider = yield take(channel)
+    return txHashesToProvider
+  } catch (error) {
+    Logger.error(TAG + 'fetchTxHashesToProviderMapping', error)
+  } finally {
+    if (yield cancelled()) {
+      channel.close()
+    }
+  }
+}
+
+export function* tagTxsWithProviderInfo({ transactions }: NewTransactionsInFeedAction) {
```

Comment:
The way this tagging was working was here in the saga we were listening for changes in the `registrations/${address}/txHashes` node and when we found a new one we stored the display info for this in Redux. Then, from the UI we just used what was stored in Redux to show it.

With this change we are only checking when there is a new transaction. So if a user first sees the transaction in the feed and we _then_ receive the webhook the transaction won't be labeled properly even though the value will be stored in Firebase (because we only check it when we see the transaction for the first time).

Not a big deal because this is unlikely to happen very often, hopefully the webhook and the transaction going through happen at the same time so when the user opens the app the labeling info will be ready.

Also, like we discussed offline I think it would be better for all this labeling to happen server-side which will give us greater backwards-compatibility when we make changes and will keep the client-side logic simpler. Just mentioning this because it's something I noticed.

Reply:
That's a good call out, thanks for pointing it out. Not ideal but hopefully that doesn't happen frequently in practice.

Definitely agree that refactoring is needed in the transaction labeling. It's done in a very disjoint way that makes the client-side code unnecessarily complicated. I'd be supportive of raising this as a potential task to prioritize.